mars/tensor/expressions/reshape/reshape.py

```diff
@@ -154,6 +157,40 @@ def _reduce_nsplit_by_expand(splits, expand):
                 out.append(good)
         return tuple(out)
 
+    @staticmethod
+    def _tile_as_shuffle(op):
+        in_tensor = op.input
+        tensor = op.outputs[0]
+        new_shape = op.newshape
+        shuffle_inputs, shuffle_outputs = [], []
+        axis_offsets = [[0] + np.cumsum(ns)[:-1].tolist() for ns in in_tensor.nsplits]
+
+        chunk_size_tuples = tuple((max(tp), len(tp)) for tp in in_tensor.nsplits)
+        max_chunk_size = max(tp[0] for tp in chunk_size_tuples)
+        chunk_size = tuple(tp[0] if tp[1] > 1 else max_chunk_size
+                           for tp in chunk_size_tuples)
+        out_nsplits = decide_chunk_sizes(new_shape, chunk_size, tensor.dtype.itemsize)
```

Comment:
If the reshaped tensor has different `ndim` with the input tensor, some errors may occur here.